Templates/UnifiedMultiplayerGem/Template/Code/CMakeLists.txt

```diff
@@ -0,0 +1,298 @@
+    BUILD_DEPENDENCIES
+        PUBLIC
+            AZ::AzCore
+            AZ::AzFramework
+            Gem::Multiplayer.Client
+        PRIVATE
+            Gem::Multiplayer.Client.Static
+)
+
+ly_add_target(
+    NAME ${gem_name}.Server.Private.Object STATIC
+    NAMESPACE Gem
+    FILES_CMAKE
+        ${NameLower}_private_files.cmake
+        ${pal_dir}/${NameLower}_private_files.cmake 
+    TARGET_PROPERTIES
+        O3DE_PRIVATE_TARGET TRUE
+    INCLUDE_DIRECTORIES
+        PRIVATE
+            Include
+            Source
+    BUILD_DEPENDENCIES
+        PUBLIC
+            AZ::AzCore
+            AZ::AzFramework
+            Gem::Multiplayer.Server
+        PRIVATE
+            Gem::Multiplayer.Server.Static
+)
+
+ly_add_target(
+    NAME ${gem_name}.Unified.Private.Object STATIC
+    NAMESPACE Gem
+    FILES_CMAKE
+        ${NameLower}_private_files.cmake
+        ${pal_dir}/${NameLower}_private_files.cmake 
+    TARGET_PROPERTIES
+        O3DE_PRIVATE_TARGET TRUE
+    INCLUDE_DIRECTORIES
+        PRIVATE
+            Include
+            Source
+    BUILD_DEPENDENCIES
+        PUBLIC
+            AZ::AzCore
+            AZ::AzFramework
+            Gem::Multiplayer
+        PRIVATE
+            Gem::Multiplayer.Unified.Static
+)
+
+# Here add ${gem_name} target, it depends on the Private Object library and Public API interface
+ly_add_target(
+    NAME ${gem_name}.Client ${PAL_TRAIT_MONOLITHIC_DRIVEN_MODULE_TYPE}
+    NAMESPACE Gem
+    FILES_CMAKE
+        ${NameLower}_shared_files.cmake
+        ${pal_dir}/${NameLower}_shared_files.cmake
+    INCLUDE_DIRECTORIES
+        PUBLIC
+            Include
+        PRIVATE
+            Source
+    BUILD_DEPENDENCIES
+        PUBLIC
+            Gem::${gem_name}.API
+        PRIVATE
+            Gem::${gem_name}.Client.Private.Object
+)
+
+ly_add_target(
+    NAME ${gem_name}.Server ${PAL_TRAIT_MONOLITHIC_DRIVEN_MODULE_TYPE}
+    NAMESPACE Gem
+    FILES_CMAKE
+        ${NameLower}_shared_files.cmake
+        ${pal_dir}/${NameLower}_shared_files.cmake
+    INCLUDE_DIRECTORIES
+        PUBLIC
+            Include
+        PRIVATE
+            Source
+    BUILD_DEPENDENCIES
+        PUBLIC
+            Gem::${gem_name}.API
+        PRIVATE
+            Gem::${gem_name}.Server.Private.Object
+)
+
+ly_add_target(
+    NAME ${gem_name} ${PAL_TRAIT_MONOLITHIC_DRIVEN_MODULE_TYPE}
+    NAMESPACE Gem
+    FILES_CMAKE
+        ${NameLower}_shared_files.cmake
+        ${pal_dir}/${NameLower}_shared_files.cmake
+    INCLUDE_DIRECTORIES
+        PUBLIC
+            Include
+        PRIVATE
+            Source
+    BUILD_DEPENDENCIES
+        PUBLIC
+            Gem::${gem_name}.API
+        PRIVATE
+            Gem::${gem_name}.Unified.Private.Object
+)
+
+# By default, we will specify that the above target ${gem_name} would be used by
+# Client and Server type targets when this gem is enabled.  If you don't want it
+# active in Clients or Servers by default, delete one of both of the following lines:
```

Comment:
Does this sentence need to be updated/removed?

Reply:
It can stay. It's valid to want Client and Unified but not Server for example.